packages/venia-concept/generateGraphQlFragmentTypes.js

```diff
@@ -0,0 +1,43 @@
+const fetch = require('node-fetch');
+const fs = require('fs');
+
+// Right now I've hardcoded this url because we only have process.env.MAGENTO_BACKEND_URL on run time
+fetch('https://staging-5em2ouy-mfwmkrjfqvbjk.us-4.magentosite.cloud/graphql', {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({
+        variables: {},
+        query: `
+      {
+        __schema {
+          types {
+            kind
+            name
+            possibleTypes {
+              name
+            }
+          }
+        }
+      }
+    `
+    })
+})
+    .then(result => result.json())
+    .then(result => {
+        // here we're filtering out any type information unrelated to unions or interfaces
+        const filteredData = result.data.__schema.types.filter(
+            type => type.possibleTypes !== null
+        );
+        result.data.__schema.types = filteredData;
+        fs.writeFile(
```

Comment:
If we move the logic to buildpack we could inline the fragment types into the bundle using the `DefinePlugin` or similar to avoid having to read a file at run time.